### PR TITLE
LPD-37074 2

### PR DESCRIPTION
**Pull request by vngoliveira**

### CODE REVIEW

**Comment by vngoliveira** [comment]
ci:test:sf

**Comment by vngoliveira** [comment]
ci:test:relevant

**Comment by vngoliveira** [comment]
ci:test:playwright

**Comment by vngoliveira** [comment]
ci:test:sf

**Comment by vngoliveira** [comment]
ci:test:sf

**Comment by vngoliveira** [comment]
ci:test:sf

File: modules/test/playwright/tests/dynamic-data-mapping-form-web/lxc-environment/formDataProvider.spec.ts [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+);
+
+let hasDataProvider: boolean = false;
+
+const DEFAULT_VIRTUAL_INSTANCE_NAME = 'www.able.com';
+
+test.afterEach(async ({formsPage, page}) => {
+	await formsPage.goTo();
+
+	await deleteItems(formsPage, page);
```
Comment (cgmonte) [inline modules/test/playwright/tests/dynamic-data-mapping-form-web/lxc-environment/formDataProvider.spec.ts:31]: This is not part of this PR, but I will add a comment about anyway: (almost) every `ComponentPage` is instantiated with a `new ComponentPage(page)`, so it already has a reference to Playwrights page instance. This `deleteItems` util does not need to receive two arguments. This util could be:

```
import {FormsPage} from '../../../pages/dynamic-data-mapping-form-web/FormsPage';

export async function deleteItems(formsPage: FormsPage) {
	await formsPage.page.waitForTimeout(1000);

	if (formsPage.managementToolbarSelectAllItems.isEnabled()) {
		await formsPage.managementToolbarSelectAllItems.click();

		formsPage.page.once('dialog', (dialog) => {
			dialog.accept();
		});

		await formsPage.managementToolbarDeleteButton.click();
	}
}
```
You don't need to change this now. I have created a [technical debit](https://liferay.atlassian.net/browse/LPD-39111) for this.

File: modules/test/playwright/tests/dynamic-data-mapping-form-web/lxc-environment/formDataProvider.spec.ts [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+			.getByLabel('Select from List', {exact: true})
+			.click();
+
+		await expect(
+			formPreviewPage.getByRole('option', {name: 'Algeria'})
+		).toBeVisible();
+
+		await formPreviewPage.close();
+
+		await performLogout(newPage);
```
Comment (cgmonte) [inline modules/test/playwright/tests/dynamic-data-mapping-form-web/lxc-environment/formDataProvider.spec.ts:105]: Is this necessary? After the test ends, the browser instance is ended with all tokens, auth cookies etc. We use `performLogout` when the test needs to log out. Do we need it?

Comment (cgmonte) [reply modules/test/playwright/tests/dynamic-data-mapping-form-web/lxc-environment/formDataProvider.spec.ts:105]: Maybe we cannot delete the virtual instance with a logged in user? :thinking:

File: modules/test/playwright/tests/dynamic-data-mapping-form-web/lxc-environment/formDataProvider.spec.ts [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+		await newPage.close();
+
+		liferayConfig.environment.baseUrl = defaultBaseUrl;
```
Comment (cgmonte) [inline modules/test/playwright/tests/dynamic-data-mapping-form-web/lxc-environment/formDataProvider.spec.ts:109]: From this line on, we are cleaning up. With the test fails before this line, it wouldn't run this code and the virtual instance would be kept with the modified baseUrl. Please move this to an `afterEach` .

File: modules/test/playwright/tests/dynamic-data-mapping-form-web/lxc-environment/formDataProvider.spec.ts [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+			'Form with data provider on virtual instance'
+		);
+
+		const formBuilderPage = new FormBuilderPage(newPage);
+
+		const pagePromise = newPage.waitForEvent('popup');
+
+		await formBuilderPage.openFormSubmission();
+
+		const formPreviewPage = await pagePromise;
```
Comment (cgmonte) [inline modules/test/playwright/tests/dynamic-data-mapping-form-web/lxc-environment/formDataProvider.spec.ts:93]: Is this the preview? I think it is the real form, not the preview. If so, rename `formPreviewPage` to something that better reflects the actual page we are in.

**Review by cgmonte — CHANGES_REQUESTED** [review]
Hi @vngoliveira! That was great teamwork! I hope I can make some contributions too by recommending changes to this PR. Please let me know your thoughts on my comments. Thanls!

File: modules/test/playwright/tests/dynamic-data-mapping-form-web/lxc-environment/formDataProvider.spec.ts (hunk 2 above)
Comment (vngoliveira) [reply modules/test/playwright/tests/dynamic-data-mapping-form-web/lxc-environment/formDataProvider.spec.ts:105]: I was following the scim.spec.ts test, but apparently it is not necessary.

**Comment by vngoliveira** [comment]
ci:test:sf

**Comment by vngoliveira** [comment]
ci:test:relevant

**Comment by vngoliveira** [comment]
ci:test:playwright

**Comment by vngoliveira** [comment]
ci:test:sf

**Comment by vngoliveira** [comment]
ci:test:relevant

**Comment by vngoliveira** [comment]
ci:test:playwright

**Comment by vngoliveira** [comment]
ci:test:sf

**Comment by vngoliveira** [comment]
ci:test:relevant

**Comment by vngoliveira** [comment]
ci:test:playwright

**Comment by vngoliveira** [comment]
ci:test:relevant

**Comment by vngoliveira** [comment]
ci:test:playwright

**Comment by vngoliveira** [comment]
ci:reevaluate:1287401_11395

**Comment by vngoliveira** [comment]
ci:report

**Comment by vngoliveira** [comment]
ci:report:41582550

**Comment by vngoliveira** [comment]
ci:report:41572956

**Comment by vngoliveira** [comment]
Hey team!
Based on the CI:report for the relevant, there aren't unique failures related to my changes; for the playwright, the reports appointed two unique: formSubmission.spec.ts and formView.spec.ts.
The first one passes locally, but the second fails locally and in the CI with an API error, and this error is not related to the changes made in this PR.
So, I am sending this PR manually to BChan.

**Pull request closed without merging**